clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -95,10 +101,10 @@ public void needReassignment() {
     }
 
     public void subscribe(TopicPartition tp) {
-        if (this.subscribedTopics.size() > 0)
-            throw new IllegalStateException("Subcription to topics and partitions are mutually exclusive");
+        if (!this.subscribedTopics.isEmpty())
+            throw new IllegalStateException("Subscription to topics and partitions are mutually exclusive");
         this.subscribedPartitions.add(tp);
-        this.assignedPartitions.add(tp);
+        this.assignedPartitions.put(tp, new TopicPartitionState());
```

Comment:
Do you think an overload of put, i.e., put(tp) will be better than creating and passing an object which this method does not keep track of or need?